plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/DiagramToXtextCommands.java

```diff
@@ -473,37 +495,64 @@ private static void clearModel(Model domainModel) {
       domainModel.getElements().clear();
     }
   }
-  
+
   private static Alt createAlt(TextualScenarioFactory factory, InteractionOperand operand) {
     Alt alt = factory.createAlt();
     alt.setKeyword(DslConstants.ALT);
+    alt.setOver(DslConstants.OVER);
     EList<InstanceRole> coveredInstanceRoles = operand.getCoveredInstanceRoles();
-    for(InstanceRole ir : coveredInstanceRoles) {
+    for (InstanceRole ir : coveredInstanceRoles) {
       alt.getTimelines().add(ir.getName());
     }
     alt.setCondition(getConditionText(operand));
     return alt;
   }
-  
+
   private static Block addAltBlock(TextualScenarioFactory factory, Alt alt, InteractionOperand operand) {
     ElseBlock elseBlock = factory.createElseBlock();
     elseBlock.setCondition(getConditionText(operand));
     alt.getElseBlocks().add(elseBlock);
-    
+
     Block altBlock = createBlock(factory);
     elseBlock.setBlock(altBlock);
-    
+
     return altBlock;
   }
-  
+
   private static Block createBlock(TextualScenarioFactory factory) {
     Block block = factory.createBlock();
     block.setBegin("{");
     block.setEnd("}");
-    
+
     return block;
   }
-  
+
+  private static StateFragment createStateFragment(TextualScenarioFactory factory, Object fragment, String name) {
+    StateFragment stateFragment = factory.createStateFragment();
+
+    AbstractState stateMode = ((org.polarsys.capella.core.data.interaction.impl.StateFragmentImpl) fragment)
+        .getRelatedAbstractState();
+    AbstractFunction function = ((org.polarsys.capella.core.data.interaction.impl.StateFragmentImpl) fragment)
+        .getRelatedAbstractFunction();
+    if (stateMode != null) {
```

Comment:
use StateFragment instead of StateFragmentImpl.
also for the cast, the package is already imported.
cast with short name: AbstractFunction function = ((StateFragment) fragment)
        .getRelatedAbstractFunction();

In general use the interface rather than the implementation.